clients/src/main/java/org/apache/kafka/common/protocol/Message.java

```diff
@@ -96,6 +98,42 @@
      */
     Struct toStruct(short version);
 
+    /**
+     * Reads this message from a Jackson JsonNode object.  This will overwrite
+     * all relevant fields with information from the Struct.
+     *
+     * For the most part, we expect every JSON object in the input to be the
+     * correct type.  There is one exception: we will deserialize numbers
+     * represented as strings.  If the numeric string begins with 0x, we will
+     * treat the number as hexadecimal.
+     *
+     * Note that we expect to see NullNode objects created for null entries.
+     * Therefore, please configure your Jackson ObjectMapper with
+     * setSerializationInclusion({@link JsonInclude.Include#ALWAYS}).
+     * Other settings may silently omit the nulls, which is not the
+     * semantic that Kafka RPC uses.  (Including a field and setting it to
+     * null is different than not including the field.)
+     *
+     * @param node          The source node.
+     * @param version       The version to use.
+     */
```

Comment:
Needs `@throws {@see org.apache.kafka.common.errors.UnsupportedVersionException}` (so does `fromStruct()`, BTW -- it is missing).

Reply:
added